From stern@rowland.harvard.edu  Tue Jun  2 12:19:45 2009
From: Alan Stern <stern@rowland.harvard.edu>
Date: Thu, 21 May 2009 17:37:58 -0400 (EDT)
Subject: USB: usb-storage: fix return values from init functions
To: Greg KH <greg@kroah.com>, Matthew Dharm <mdharm-usb@one-eyed-alien.net>
Cc: USB Storage list <usb-storage@lists.one-eyed-alien.net>
Message-ID: <Pine.LNX.4.44L0.0905211730330.4783-100000@iolanthe.rowland.org>


This patch (as1242) fixes the return values from the special
init functions in usb-storage.  They are supposed to return 0 for
success, not USB_STOR_TRANSPORT_GOOD.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/storage/option_ms.c |    8 ++++----
 drivers/usb/storage/sierra_ms.c |    2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/usb/storage/option_ms.c
+++ b/drivers/usb/storage/option_ms.c
@@ -96,7 +96,7 @@ int option_ms_init(struct us_data *us)
 	    udev->descriptor.bDeviceSubClass != 0 ||
 	    udev->descriptor.bDeviceProtocol != 0 ||
 	    udev->actconfig->desc.bNumInterfaces == 3)
-		return USB_STOR_TRANSPORT_GOOD;
+		return 0;
 
 	US_DEBUGP("Option MS: option_ms_init called\n");
 
@@ -107,7 +107,7 @@ int option_ms_init(struct us_data *us)
 	    iface_desc->desc.bInterfaceProtocol != 0x50) {
 		US_DEBUGP("Option MS: mass storage interface not found, no action "
 		          "required\n");
-		return USB_STOR_TRANSPORT_GOOD;
+		return 0;
 	}
 
 	/* Find the mass storage bulk endpoints */
@@ -127,7 +127,7 @@ int option_ms_init(struct us_data *us)
 	if (!ep_in_size || !ep_out_size) {
 		US_DEBUGP("Option MS: mass storage endpoints not found, no action "
 		          "required\n");
-		return USB_STOR_TRANSPORT_GOOD;
+		return 0;
 	}
 
 	/* Force Modem mode */
@@ -143,6 +143,6 @@ int option_ms_init(struct us_data *us)
 		          " requests it\n");
 	}
 
-	return USB_STOR_TRANSPORT_GOOD;
+	return 0;
 }
 
--- a/drivers/usb/storage/sierra_ms.c
+++ b/drivers/usb/storage/sierra_ms.c
@@ -202,6 +202,6 @@ int sierra_ms_init(struct us_data *us)
 complete:
 	result = device_create_file(&us->pusb_intf->dev, &dev_attr_truinst);
 
-	return USB_STOR_TRANSPORT_GOOD;
+	return 0;
 }
 
